src/node_buffer.cc

```diff
@@ -700,6 +927,38 @@ void IndexOfNumber(const FunctionCallbackInfo<Value>& args) {
       ptr ? static_cast<int32_t>(ptr_char - obj_data) : -1);
 }
 
+void IsNull(const FunctionCallbackInfo<Value>& args) {
+  ASSERT(args[0]->IsObject());
+  ARGS_THIS(args[0].As<Object>())
+
+  int64_t offset = args[1]->IntegerValue();
+  char *ptr = obj_data + offset;
+
+  args.GetReturnValue().Set( ptr == NULL );
+}
+
+void Address(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  ASSERT(args[0]->IsObject());
+  ARGS_THIS(args[0].As<Object>())
```

Comment:
This is dangerous. `ARGS_THIS` assumes you know that the `Object` has external array data, and it's easy enough to hack around this in JS to cause an abort. Always run `HasInstance()` on the object to verify this. If not then throw.